### PR TITLE
Bug fixes and enhancement

### PR DESCRIPTION
**Pull request by georgolden**
- [x] tests and linter show no problems (`npm t`)
- [x] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

### CODE REVIEW

**Comment by georgolden** [comment]
I don't know why windows-latest tests failed. I checked on windows with 1000 times npm run test and no one failed test. Maybe there is different version of metautil work in CI?

File: lib/utilities.js [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 const path = require('path');
 
+const UPPER_LETTERS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+const LOWER_LETTERS = 'abcdefghijklmnopqrstyvwxyz';
```
Comment (tshemsedinov) [inline lib/utilities.js:6]: I'd prefer to compare `charCodeAt` with `a` and `z` instead of finding char in strings `UPPER_LETTERS` and `LOWER_LETTERS`

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

File: lib/utilities.js (hunk 1 above)
Comment (georgolden) [reply lib/utilities.js]: Done

**Comment by georgolden** [comment]
@tshemsedinov please land. Hardly needed for new Schema

File: lib/utilities.js [hunk 2]
```diff
@@ -79,7 +79,13 @@ const split = (s, separator) => {
   return [s.slice(0, i), s.slice(i + separator.length)];
 };
 
-const isFirstUpper = (s) => !!s && s[0] === s[0].toUpperCase();
+const inRange = (num, start, end) => num <= end && num >= start;
```
Comment (tshemsedinov) [inline lib/utilities.js:82]:
```suggestion
const inRange = (x, min, max) => x >= min && x <= max;
```

File: lib/utilities.js [hunk 3]
```diff
@@ -79,7 +79,13 @@ const split = (s, separator) => {
...
+
+const isFirstUpper = (s) => !!s && inRange(s.charCodeAt(0), 65, 90);
+
+const isFirstLower = (s) => !!s && inRange(s.charCodeAt(0), 97, 122);
```
Comment (tshemsedinov) [inline lib/utilities.js:86]:
```suggestion

const isFirstUpper = (s) => !!s && inRange(s[0], 'A', 'Z');

const isFirstLower = (s) => !!s && inRange(s[0], 'a', 'z');
```

Comment (tshemsedinov) [reply lib/utilities.js:86]: Constants are better then magic numbers but letter are much better.

**Review by tshemsedinov — APPROVED** [review]

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/metautil/commit/541a42484c1b61138d06d345f112b86e87f7ad85, https://github.com/metarhia/metautil/commit/ac83a75619bdc5466730590d7ac553ff3b3b434b, and https://github.com/metarhia/metautil/commit/13a25bb145ee937c774bf11cbe5d4f9980f4e476

**Pull request closed without merging**